testdriver/testplan.py

```diff
@@ -4,519 +4,497 @@
+            if self.debug:
+                logging.info('++++++ Results file path = %s', self.outputFilePath)
+                self.resultsFile = open(self.outputFilePath, encoding='utf-8', mode='w')
+        except BaseException as error:
+            print('*** Cannot open results file at %s. Err = %s' %
+                  (self.outputFilePath, error))
+            self.resultsFile = open(self.outputFilePath, encoding='utf-8', mode='w')
+
+        # Store information the test run
+        # TODO: remove!??
+        self.generate_header()
+
+        # For each test
+        #   send JSON data --> executor via stdout
+        #      with subprocess
+        #      handle any exceptions
+        #   retrieve stdout result
+        #   add format for JSON item
+        #   output to result file
+        #   output to result file
+
+        # Work with the executor, running N tests each invocation
+        try:
+            per_execution = int(self.options.per_execution)
+        except TypeError:
+            per_execution = 1
+        num_errors = self.run_all_single_tests(per_execution)
+
+        env_dict = {}
+        try:
+            env_string = self.options.environment
+            env_options = env_string.split(';')
+            # Set the environment from the options, each separated with '='
+            for option in env_options:
+                parts = option.split('=')
+                env_dict[parts[0]] = parts[1]
+                # The environment variables for running the command line.
+            self.exec_env = env_dict
+        except (AttributeError, KeyError):
+            env_dict = None
+        # TODO: Use env_dict for environment information.
+
+        # Complete outputFile
+        self.complete_output_file(num_errors)
+        return  # TODO: status
+
+    def run_all_single_tests(self, tests_per_execution=1):
+        # Print out each "test" item as a single line of text
+        # Returns dictionary of all test results
+        num_errors = 0
+
+        # Save all results in single JSON output.
+        all_test_results = []
+
+        test_count = len(self.tests)
+        test_num = 0
+        lines_in_batch = 0
+
+        # A group of tests separated by newlines
+        test_lines = []
+
+        # N tests may be given to send_one_line in a single batch.
+        formatted_count = '{:,}'.format(test_count)
+        for test in self.tests:
+            test.update({"test_type": self.testScenario})
+
+            if self.progress_interval and test_num % self.progress_interval == 0:
+                formatted_num = '{:,}'.format(test_num)
+                print('Testing %s / %s. %s of %s' % (
+                    self.exec_list[0], self.testScenario, formatted_num, formatted_count), end='\r')
+
+            # Accumulate tests_per_execution items into a single outline
+            if lines_in_batch < tests_per_execution:
+                test_lines.append(json.dumps(test))
+                lines_in_batch += 1
```

Comment:
Observation: I don't immediately see where these lines had been before; they seem to be newly added

Reply:
These were changed from the previous version, removing an intermediate variable.